Engine/ac/string.cpp

```diff
@@ -49,6 +49,67 @@ int String_IsNullOrEmpty(const char *thisString)
     return 0;
 }
 
+void calculate_joint_length(const size_t count, const uint32_t *&handles, int sep_len, int sep_ulen, int &total_len, int &total_ulen)
+{
+    total_len = 0;
+    total_ulen = 0;
+    for (uint32_t i = 0; i < count; ++i)
+    {
+        auto address = ccGetObjectAddressFromHandle(handles[i]);
+        if (address != nullptr) {
+            auto const& sh = ScriptString::GetHeader(address);
+            total_len += sh.Length;
+            total_ulen += sh.ULength;
+        }
+        if (i != count - 1) {
```

Comment:
I think you may simplify this by not adding separator lengths in a loop, but add them once afterwards, since this only depends on the `count`:
```
total_len += sep_len * (count - 1);
total_ulen += sep_ulen * (count - 1);
```

Reply:
that's true, I didn't notice that when writing the code.